### PR TITLE
Add some more Spectra methods

### PR DESCRIPTION
**Pull request by jotsetung**
Add `filterMsLevel,Spectra`, `coerce,Spectra,list` and `coerce,Spectra,MSnExp` (issue #370) and `clean,Chromatograms` (issue #354).

### CODE REVIEW

File: R/DataClasses.R [hunk 1]
```diff
@@ -523,9 +523,9 @@ setClass("ProcessingStep",
 #'     example, a \code{Chromatogram} for a SRM transition 273 -> 153 will have
 #'     a \code{@precursorMz = c(273, 273)} and a
 #'     \code{@productMz = c(153, 153)}.
-#' 
+#'
```
Comment (sgibb) [inline R/DataClasses.R]: JFTR: first result of the [EditorConfig](https://editorconfig.org) plugin. As I mentioned in #376 these additional spaces at the line ending happened to me all the time (and I always use `git add -p` to exclude them from the commit). In a few commits we would have edited all files and won't be annoyed by trailing whitespaces anymore. Alternatively we could run `for i in *.R; do sed -i 's#[[:space:]]*$##g' ${i}; done` to get rid of them in just one commit.

Comment (jotsetung) [reply R/DataClasses.R]: I don't mind if they get fixed successively. I like your editorconfig addon 👍

**Comment by jotsetung** [comment]
No change requests at all @sgibb ? for real?

File: R/methods-Spectra.R [hunk 2]
```diff
@@ -514,6 +523,44 @@ setMethod("combineSpectra", "Spectra", function(object, fcol,
...
+    if (!length(names(from)))
+        names(from) <- as.character(seq_len(length(from)))
+    if (is.null(mcols(from)))
+        fd <- AnnotatedDataFrame(data.frame(spectrum = seq_len(length(from)),
+                                            row.names = names(from)))
+    else
+        fd <- AnnotatedDataFrame(as.data.frame(mcols(from)))
+    fromf <- as.character(unique(fromFile(from)))
+    process <- new("MSnProcess", files = fromf,
+                   processing = paste("Data converted from Spectra:", date()))
```
[the remark this answers is not in the document]
Comment (jotsetung) [reply R/methods-Spectra.R]: I wasn't even aware of the `"[", date(), "]"` - I used `readMSData` as *template*.

File: R/methods-MSnExp.R [hunk 3]
```diff
@@ -565,3 +565,15 @@ setAs("MSnExp", "data.frame", function(from) {
...
+
+setAs("MSnExp", "Spectra", function(from) {
+    fdta <- fData(from)
+    red_cn <- c("fileIdx", "spIdx", "smoothed", "seqNum", "acquisitionNum",
+                "msLevel", "polarity", "originalPeaksCount", "totIonCurrent",
+                "retentionTime", "basePeakMZ", "basePeakIntensity",
+                "collisionEnergy", "ionisationEnergy", "precursorScanNum",
+                "precursorMZ", "precursorCharge", "precursorIntensity",
+                "mergedScan", "centroided", "spectrum")
+    fdta <- fdta[, !colnames(fdta) %in% red_cn, drop = FALSE]
```
Comment (lgatto) [inline R/methods-MSnExp.R]: Why aren't all feature variables passed along during coercion? I would prefer to keep all of them by default and have a `selectFeatureData` function to select those to keep.

Comment (lgatto) [reply R/methods-MSnExp.R]: Indeed, thank you. Is there already an accessor for these (`centroided,Spectra`, ...)? The implementation will be identical to the corresponding in memory `MSnExp` methods.

Comment (jotsetung) [reply R/methods-MSnExp.R]: Yes, all methods to access slot values for `Spectrum` objects are also implemented for `Spectra` objects (in *methods-Spectra.R*), returning always a vector of length equal to `Spectra` with the corresponding slot values.

**Pull request merged**